pkg/channel/http/http_channel.go

```diff
@@ -68,13 +68,16 @@ type Channel struct {
 // CreateLocalChannel creates an HTTP AppChannel
 //
 //nolint:gosec
-func CreateLocalChannel(port, maxConcurrency int, pipeline httpMiddleware.Pipeline, spec config.TracingSpec, sslEnabled bool, maxRequestBodySizeMB, readBufferSizeKB int) (channel.AppChannel, error) {
+func CreateLocalChannel(port, maxConcurrency int, pipeline httpMiddleware.Pipeline, spec config.TracingSpec, sslEnabled bool, maxRequestBodySizeMB, readBufferSizeKB int, allowInsecureTLS bool) (channel.AppChannel, error) {
```

Comment:
Not gonna block the PR due to this, but if you get a chance can you please convert this interface to accept a single struct with options, rather than 8 arguments? It becomes hard to even read what the invocations are doing.

Reply:
Good idea, let me refactor in this PR itself.